ospd_openvas/daemon.py

```diff
@@ -1558,201 +1178,72 @@ def exec_scan(self, scan_id: str):
             self.add_scan_error(
                 scan_id,
                 name='',
-                host=target,
+                host='',
                 value=(
                     'It was not possible to start the scan,'
                     'because a pending feed update. Please try later'
                 ),
             )
             return 2
 
-        ports = self.get_scan_ports(scan_id)
+        do_not_launch = False
+        kbdb = self.main_db.get_new_kb_database()
+        scan_prefs = PreferenceHandler(scan_id, kbdb, self.scan_collection)
+        scan_prefs.set_target()
+        ports = scan_prefs.set_ports()
```

Comment:
A set function/method should *usually* not return a value and expect some parameters.